laravel/tests/Unit/AutoAssignmentTest.php

```diff
@@ -305,16 +319,16 @@ public function testAnestDoubleAssignedCCCT() {
 
         $bGotAnest = self::correctAssignments($expectedAssignments1);
         $cGotAnest = self::correctAssignments($expectedAssignments2);
-        $this->assertTrue( $bGotAnest || $cGotAnest);
-
+        
+        self::deleteOptionsInDatabase($optionIds);
         if($bGotAnest){
-            $this->assertTrue(self::correctProbTotals($expectedProbTotals1));
             self::deleteExpectedAssignments($expectedAssignments1);
+            $this->assertTrue(self::correctProbTotals($expectedProbTotals1));
         } else{
-            $this->assertTrue(self::correctProbTotals($expectedProbTotals2));
             self::deleteExpectedAssignments($expectedAssignments2);
+            $this->assertTrue(self::correctProbTotals($expectedProbTotals2));
         }
```

Comment:
Shouldn't we always (outside of the `if else`) delete the actual assignments that were created, rather than conditionally deleting expected values?

Sidenote: Ideally, we'd just use [the `RefreshDatabase` trait](https://laravel.com/docs/8.x/database-testing#resetting-the-database-after-each-test), but since test/dev share a database (bad choice imo, but out of our hands), we can't :frowning_face:.

Reply:
I put it in an if else to have the `deleteExpectedAssignments()` be called once rather than call it twice outside the if else and basically duplicate the delete logic.

Logically it makes sense to me -> delete the assignments that were actually created (this is variable: either ExpectedAssignments1 are created or ExpectedAssignments2 are created)

Reply:
It seems we want to be able to delete the assignments that are **actually** created, rather than the **expected** assignments.

Unless I'm missing something, it seems possible that additional erroneous assignments could be created, or that the created assignments could be only partially correct. In either case, the test seems it would fail to clean up correctly.

Reply:
Did you have some solution in mind? The assignments are created during the AutoAssignment method call so there really is no way to know exactly which assignments it created. Since we can't use `RefreshDatabase ` this was the best work around I could come up with.

Reply:
The ideal solution would be seeing if we can get a third database for testing. That way we can use the refresh database trait.

If we can't do that, or it's too much work, we could make a note of the greatest ID at the beginning of the test and at the end of the test delete all assignments with an ID greater than that.

Reply:
The point of this PR was just to move the asserts after the db stuff. If we want to do anything further, we can probably create a separate task for it. The logic is the same as in other tests that work with the db. Since we have higher priorities right now and no new logic was added in this pr, @cj81499 would you consider revisiting this in a later task?

Reply:
sure! approving now.

Reply:
I updated the description of the card on our backlog regarding testing so we don't forget about this.